pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -722,11 +722,7 @@ impl InMemoryLayer {
                 before_segsizes.insert(*lsn, *size);
             }
         }
-        let mut before_page_versions = inner.page_versions.iter().filter(|tup| {
-            let ((_blknum, lsn), _pv) = tup;
-
-            *lsn < end_lsn
-        });
+        let mut before_page_versions = inner.page_versions.ordered_page_version_iter(Some(end_lsn));
```

Comment:
The order going into DeltaLayer doesn't matter. At the moment it will all go into a BTreeMap, which will take care of the order for us. I'm just using this pre-sorted setup now to avoid changing `DeltaLayer` code much. In the future, I will come up with a more customized routine for efficiently going from `PageVersions` to `VecMap<(u32, Lsn), PageVersion>` (once we use VecMap in DeltaLayer)